src/reflect/gclayout_precise.go

```diff
@@ -0,0 +1,37 @@
+//go:build gc.precise
+
+package reflect
+
+import "unsafe"
+
+var (
+	// Constants for use with alloc()
+	// See runtime/gc_precise.go
+	// pppppppp_pppppppp_pppppppp_ppsssss1
+	gcLayoutNoPtrs  = unsafe.Pointer(uintptr(0b0000_00001_1))
+	gcLayoutPointer = unsafe.Pointer(uintptr(0b0001_00001_1))
+	gcLayoutString  = unsafe.Pointer(uintptr(0b0001_00010_1))
+	gcLayoutSlice   = unsafe.Pointer(uintptr(0b0001_00011_1))
+)
+
+func (r *rawType) gcLayout() unsafe.Pointer {
+
+	if r.Kind() < String {
+		return gcLayoutNoPtrs
+	}
+
+	if r.Kind() == Pointer || r.Kind() == UnsafePointer {
+		return gcLayoutPointer
+	}
```

Comment:
Channels and maps are also pointers, and can be added here.
Also, you could consider doing something like this:

```go
kind := r.Kind()
if kind < String {
    return gcLayoutNoPtrs
}
switch kind {
case Pointer, UnsafePointer, Map, Chan:
    return gcLayoutPointer
case String:
    // etc
}
```